src/dotnet/UI.Blazor/Components/ContextMenu/BlazorContextMenu/Components/Item.razor

```diff
@@ -0,0 +1,178 @@
+            string classes;
+            if (Enabled) {
+                classes = Helpers.AppendCssClasses((OverrideDefaultCssClass ?? template.DefaultCssOverrides.MenuItemCssClass),
+                                            (CssClass ?? settings.GetTemplate(Template).MenuItemCssClass));
+            }
+            else {
+                classes = Helpers.AppendCssClasses((OverrideDefaultDisabledCssClass ?? template.DefaultCssOverrides.MenuItemDisabledCssClass),
+                                            (DisabledCssClass ?? settings.GetTemplate(Template).MenuItemDisabledCssClass));
+            }
+
+            var hasSubMenu = menuTreeTraverser.HasSubMenu(this);
+            if (hasSubMenu) {
+                classes = Helpers.AppendCssClasses(classes,
+                        (OverrideDefaultCssClassWithSubMenu ?? template.DefaultCssOverrides.MenuItemWithSubMenuCssClass),
+                        (CssClassWithSubMenu ?? settings.GetTemplate(Template).MenuItemWithSubMenuCssClass));
+            }
+
+            return classes;
+        }
+    }
+
+    protected ElementReference MenuItemElement { get; set; }
+    protected string Template { get; set; } = "";
+    protected ContextMenu RootMenu { get; private set; } = null!;
+    protected ContextMenuBase ParentMenu { get; private set; } = null!;
+
+    protected override async Task OnInitializedAsync()
+    {
+        if (string.IsNullOrEmpty(Id))
+            Id = Guid.NewGuid().ToString();
+
+        var rootMenu = menuTreeTraverser.GetRootContextMenu(this);
+        RootMenu = rootMenu ?? throw new Exception("Item component must be a child of a ContextMenu component");
+
+        ParentMenu = menuTreeTraverser.GetClosestContextMenu(this)!;
+        Template = ParentMenu.GetActiveTemplate();
+        var contextMenuTrigger = RootMenu.GetTrigger();
+        var contextMenuTarget = RootMenu.GetTarget();
+
+        if (OnAppearing.HasDelegate)
+        {
+            //TODO: Remove casting when 'this' is fixed in blazor
+            var appearingArgs = new ItemAppearingEventArgs(RootMenu.Id, contextMenuTarget, contextMenuTrigger ,(Item)(object)this, Visible, Enabled, RootMenu.Data);
+            await OnAppearing.InvokeAsync(appearingArgs);
+            Visible = appearingArgs.IsVisible;
+            Enabled = appearingArgs.IsEnabled;
+        }
+    }
+
+
+    protected async Task OnClickInternal(MouseEventArgs e)
+    {
+        if (!Enabled)
+            return;
+
+        if (OnClick.HasDelegate)
+        {
+            var contextMenuTarget = RootMenu.GetTarget();
+            var contextMenuTrigger = RootMenu.GetTrigger();
+            //TODO: Remove casting when 'this' is fixed in blazor
+            var args = new ItemClickEventArgs(e, RootMenu.Id, contextMenuTarget ,contextMenuTrigger ,MenuItemElement,this, RootMenu.Data);
+            args.IsCanceled = !RootMenu.AutoHide;
+            await OnClick.InvokeAsync(args);
+            if (!args.IsCanceled)
+            {
```

Comment:
Bracket should be on the same line. Single-statement blocks shouldn't have brackets.

Reply:
done